src/arithmetic/rewrite_simplify.cc

```diff
@@ -1757,6 +1757,20 @@ Mutate_(const Variable* op, const Expr& self) {
   return self;
 }
 
+Expr RewriteSimplifier::Impl::
+Mutate_(const Cast* op, const Expr& self) {
+  Expr ret = IRMutator::Mutate_(op, self);
+  op = ret.as<Cast>();
+  // 0 and 1 are very common and useful for simplification
+  if (is_const_value(op->value, 0)) {
+    return make_const(op->type, 0);
+  }
+  if (is_const_value(op->value, 1)) {
+    return make_const(op->type, 1);
+  }
```

Comment:
could you support float(16) --> 16.000, int32(16.000) --> 16 ?

Reply:
With the implementation via `tvm::cast` cases like these should work now.